### PR TITLE
Simple omp

### PR DESCRIPTION
**Pull request by robsimmonds**
Replaces Intel TBB parallel_for loops with OpenMP pragmas. 
This code had  been run for a couple of week on the IDIA test server that provides the latest code (usually the dev branch) to IDIA users and seems to have solved a lockup problem we were seeing with the TTB based code.

### CODE REVIEW

File: Region/BasicStatsCalculator.tcc [hunk 1]
```diff
@@ -58,6 +58,27 @@ void BasicStatsCalculator<T>::operator()(const tbb::blocked_range<size_t>& r) {
...
+void BasicStatsCalculator<T>::reduce(const int start, const int end) {
+	  int i;
+#pragma omp parallel for private(i) shared(_data) reduction(min: _min_val) reduction(max:_max_val) reduction(+:_num_pixels) reduction(+:_sum) reduction(+:_sum_squares)
+    for (i = start; i < end ; i++) {
+        T val = _data[i];
+        if (std::isfinite(val)) {
+            if (val < _min_val) {
+                _min_val = val;
+            }
+			if (val > _max_val) {
```
Comment (confluence) [inline Region/BasicStatsCalculator.tcc]: Some indentation weirdness; looks like tabs instead of spaces.

**Review by confluence — CHANGES_REQUESTED** [review]
Everything looks fine except for some indentation -- it just needs a formatting pass.

**Review by veggiesaurus — CHANGES_REQUESTED** [review]
@robsimmonds some minor merge conflicts to be fixed.

@ajm-asiaa I see the ubuntu build is failing. Is it ubuntu 16.04? The version of GCC is pretty old (5.4?), and I think that's causing issues with the OpenMP parallel for

**Comment by ajm-ska** [comment]
@veggiesaurus Yes. You are correct. It is Ubuntu 16.04 using gcc version 5.4.0 20160609. I have been using 16.04 to ensure back compatibility. Maybe it is time to move to 18.04, particularly as 20.04 will be out soon. 

Meanwhile, I have fixed it for now. I just installed gcc-7 into the Ubuntu 16.04 Docker container and set it as the default the gcc. It can build successfully now.

**Comment by veggiesaurus** [comment]
> @veggiesaurus Yes. You are correct. It is Ubuntu 16.04 using gcc version 5.4.0 20160609. I have been using 16.04 to ensure back compatibility. Maybe it is time to move to 18.04, particularly as 20.04 will be out soon.

Yes, I think it makes sense to move on to supporting 18.04 and 20.04. What about releases in between? I'm on 19.10 at the moment, and things work fine, apart from the casacore repo / hdf5 library mismatch

> So please ignore that it fails on ci/circle: ubuntu for now. Will you still be able to merge it despite the checks failing?

Once the merge conflicts have been fixed, yes.

**Review by confluence — APPROVED** [review]
The formatting issues are fixed.

**Pull request merged**